### PR TITLE
Fix couple of warnings, coverage unchanged

### PR DESCRIPTION
**Pull request by cpq**

### CODE REVIEW

**Comment by mmikulicic** [comment]
Please fix tests

**Comment by mmikulicic** [comment]
Furthermore, the ansi build runs unamalgamated, and one error message in the ecma report contains the line number. Please commit after running `make run`.

**Comment by cpq** [comment]
Fixed.

File: tests/ecma_report.txt [hunk 1]
```diff
@@ -9672,7 +9672,7 @@ d-G]/.exec("a") throw SyntaxError. Actual: {"message":"[RegExp] is not defined"}
 9664	FAIL (tail -c +8805291 tests/ecmac.db|head -c 1595): [{"message":"#2: __str = new String(__obj); __str =="tostr". Actual: __str =={}"}]
 9665	FAIL (tail -c +8806887 tests/ecmac.db|head -c 1625): [{"message":"#2: function __obj(){}; __str = new String(__obj); __str =="true". Actual: __str ==[function __obj()]"}]
 9666	FAIL (tail -c +8808513 tests/ecmac.db|head -c 1051): [{"message":"#1.1: e==="intostr". Actual: e==={"message":"#1: var __obj = {toString:function(){throw "intostr"}}; __str = new String(__obj) lead throwing exception"}"}]
-9667	FAIL (tail -c +8809565 tests/ecmac.db|head -c 1059): [{"message":"#1.1: e==="invalueof". Actual: e==={"message":"to_json line 406: ast_fetch_tag(a, &var) == AST_VAR_DECL"}"}]
+9667	FAIL (tail -c +8809565 tests/ecmac.db|head -c 1059): [{"message":"#1.1: e==="invalueof". Actual: e==={"message":"to_json line 5003: ast_fetch_tag(a, &var) == AST_VAR_DECL"}"}]
```
Comment (mmikulicic) [inline tests/ecma_report.txt]: Did you rerun `make run` and commit the diff?

**Pull request merged**